refactored/cache.py

```diff
@@ -0,0 +1,56 @@
+import json
+import pickle
+
+import utils
+
+
+class Cache(object):
+    def __init__(self, traces=None, downloader_cache=None):
+        # TODO: construct from Dict[signature, List[(stack_trace, uuid)] ?
+        self.downloader_cache = downloader_cache if downloader_cache else dict()
+        self.traces = traces if traces else []
+
+    @staticmethod
+    def build(stream):
+        """build from downloaded archives"""
+
+        # TODO: is it possible to have different `signature`s for one `proto_signature` ?
```

Comment:
No, there should be only one signature for each proto_signature (that is basically a stack trace).